Feature/support webrenderer for 3.0

gguoyu opened the pull request:
# Pre-PR Checklist

- [x] I added/updated relevant documentation.
- [x] I followed the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters to submit commit message.
- [x] I squashed the repeated code commits.
- [x] I signed the [CLA].
- [x] I added/updated test cases to check the change I am making.
- [x] All existing and new tests are passing.

pba-cra commented on driver/js/packages/hippy-web-renderer/src/component/list-view.ts:
```diff
@@ -70,7 +70,7 @@ export class ListView extends HippyWebView<HTMLDivElement> {
 
   public defaultStyle() {
-    return  { display: 'flex', flexDirection: 'column', flexShrink: 0, boxSizing: 'border-box', overflow: 'scroll' };
+    return  { display: 'flex', flex: 1, flexDirection: 'column', flexShrink: 0, boxSizing: 'border-box', overflow: 'scroll' };
```
为什么要flex:1？native这边现在是flex:1吗？会不会有副作用

gguoyu replied:
这里我修改下，因为3.0 vue-next的跟节点没有设置flex:1的情况下，debug工具无法查看页面。web-renderer这边默认样式并不需要设置这个，应由业务自行设置

pba-cra commented on driver/js/packages/hippy-web-renderer/src/env/dynamic-load.ts:
```diff
@@ -18,14 +18,17 @@
-  // }
-  // ContextifyModule.LoadUntrustedContent(requestPath, encode, cb);
+export const dynamicLoad = (path: string, cb) => {
+  const s = document.createElement('script');
```
命名考虑优化一下？

gguoyu replied:
好的，我修改下

ruifanyuan committed:
fix(ios): improve image rendering performance in main thread

wwwcg committed:
fix(ios): showScrollIndicator not working on horizontal list

wwwcg committed:
docs(ios): update iOS integration guide

wwwcg committed:
fix(ios): optimize creation time of lazy-load list items

wwwcg committed:
refactor(ios): eliminate code warnings and resolve architectural issues

wwwcg committed:
fix(ios): add missing notification and remove unused variables

wwwcg committed:
refactor(ios): revert rename Hippy prefix to NativeRender

gguoyu committed:
fix(vue-next): remove vue prod devtools global sign for ios&android

wwwcg committed:
fix(ios): optimize perf of ComponentMap and support multi-rootView

wwwcg committed:
fix(ios): showsXXXScrollIndicator property of scrollview not working

siguangli2018 committed:
feat(android): add bugly app id register

siguangli2018 committed:
fix(android): add shouldNotifyNonBatchingChange for render node

wwwcg committed:
chore(ios): fix demo build problem

wwwcg committed:
chore(ios): fix demo build problem

siguangli2018 committed:
feat(android): add ENABLE_BUGLY_REPORT config

wwwcg committed:
refactor(ios): optimize local file and base64 data's loading efficiency (#3701)

* refactor(ios): optimize local file and base64 data's loading efficiency

* refactor(ios): optimize local file and base64 data's loading efficiency2

* refactor(ios): add custom userInfo in vfs completion block

In order to support third-party vfs implementations
when returning images directly, rather than raw data

Such as using SDWebImage to download resource

lvfen committed:
fix(voltron): resolve remoteServerUrl parse version id failed (#3720)

* fix(voltron): resolve remoteServerUrl parse version id failed

* fix(voltron): add license for flutter_render_test.dart

zealotchen0 committed:
feat(vue): update postcss dependencies (#3723)

* feat(vue-loader): css scoped compile to class selector

* feat(vue): update postcss dependencies

gguoyu committed:
feature(vue-next): hippy vue next ssr for 3.0 (#3539)

* feat(vue-next): commit ssr init version

* feat(vue-next): remove unused method & fix ts type issue

* feat(vue-next): optimize build file types & update doc

* feat(vue-next): ssr support commit for library & demo

* fix(vue-next): fix eslint issue

* fix(vue-next): compatible hippy 3.x node operate method & ui

* fix(vue-next): remove unused code

* fix(vue-next): fix ssr demo issue

* fix(vue-next): fix div demo issue & native background android issue

* docs(vue-next): add new vue-next ssr docs

siguangli2018 committed:
feat(android): add disallowInterceptTouchEvent for hippy view group

ilikethese committed:
fix(vfs): OnJniDelegateInvokeSync call after OnDestroyVfs

ilikethese committed:
fix(dom): add deserializer log

ilikethese committed:
fix(dom): add log for deserializer crash

ilikethese committed:
fix(layout): fix layout border default css direction

siguangli2018 committed:
fix(android): add thread check before argumentToBytes

siguangli2018 committed:
feat(android): support output render log to host

ilikethese committed:
fix(dom): remove some redunant log

wwwcg committed:
fix(ios): backgroundImage may not update in time when reusing hippyView (#3736)

* chore(ios): revert rename Hippy prefix to NativeRender

* fix(ios): backgroundImage may not update in time when reusing hippyView

replace some deprecated UIGraphics functions

* fix(ios): backgroundImage may not update in time when reusing hippyView2

wwwcg committed:
chore(ios): fix typos and update code comment (#3738)

* chore: Fix multiple typos

Signed-off-by: hugo-syn <hugo.vincent@synacktiv.com>

* chore(ios): update code comment

---------

Signed-off-by: hugo-syn <hugo.vincent@synacktiv.com>
Co-authored-by: hugo-syn <hugo.vincent@synacktiv.com>

gguoyu committed:
fix(vue-next): compatible vue patchProp parameter type change

siguangli2018 committed:
fix(android): compatible HippyArray for dispatchFunction

siguangli2018 committed:
fix(android): text node add gesture span with gesture events

wwwcg committed:
fix(ios): revert rename symbol Prefix and fix compatibility issues

wwwcg committed:
refactor(ios): improve version compatibility

Xuanquan-Zhuang committed:
feat(android): add modal features (#3680)

Co-authored-by: andrezhuang <andrezhuang@tencent.com>
Co-authored-by: OpenHippy <124017524+open-hippy@users.noreply.github.com>

wwwcg committed:
chore(ios): revert rename symbol Prefix and fix compatibility issues

zealotchen0 committed:
docs(js): add js 3.0 upgrade guidelines (#3756)

* feat(vue-loader): css scoped compile to class selector

* docs(js): add js 3.0 upgrade guidelines

* docs(js): add js 3.0 upgrade guidelines

zealotchen0 committed:
feat(vue): vue component support beforeLoadStyle disabled

siguangli2018 committed:
fix(android): should hide soft input even input not focus

iPel committed:
fix(android): fix premature text truncation (#3609)

zealotchen0 committed:
fix(vue3): delete source prop for img

zealotchen0 committed:
fix(vue3): skip patch style for same props

zealotchen0 committed:
fix(vue3): update patch style isStyleExisted function comment

wwwcg committed:
refactor(ios): normalize code and clean up some code problems

wwwcg committed:
feat(ios): add rootViewRunApplicationFinished delegate method

for HippyRootViewDelegate

wwwcg committed:
fix(ios): increase code robustness and compatibility for HippyRootView

wwwcg committed:
chore(ios): add PrivacyInfo config for iOS

wwwcg committed:
refactor(ios): minor changes for compatibility

wwwcg committed:
build(ios): fix std::function build error on Xcode15.3

wwwcg committed:
fix(ios): fix memory issue when calling JSTurboObjectWithName

wwwcg committed:
fix(ios): implementation issue with defaultSource prop of Image

siguangli2018 committed:
fix(android): setItemPrefetchEnabled to false for recycler view

wwwcg committed:
fix(ios): fix module provider not working and code refactor

siguangli2018 committed:
feat(android): add enableScrollDirectionFix for recycler view

gguoyu committed:
fix(vue-next): fix vue patchProp params type issue

gguoyu committed:
Merge remote-tracking branch 'origin/feature/support-webrenderer-for-3.0' into feature/support-webrenderer-for-3.0

siguangli2018 committed:
fix(android): concat log msg only in debug mode

gguoyu committed:
fix(vue-next): fix js release bundle size too big issue (#3779)

wwwcg committed:
fix(ios): ensure insertHippySubview is called sequentially

Some businesses depend on the invocation order of the insertHippySubview method,
so we use std::map instead to ensure that the view created first calls that method first

wwwcg committed:
feat(ios): print hippy core module logs (#3783)

* chore(ios): add necessary logs to locate problem

* feat(ios): print hippy core module logs

* feat(ios): adjust default log level on Release from Error to Info

* chore(ios): remove the duplicate core log delegate

siguangli2018 committed:
fix(android): remove ImageDataHolder recycle

wwwcg committed:
refactor(ios): unify component and module method calling mechanism

Fixed an issue where UI components does not support calling non-ID parameter type methods in hippy3

siguangli2018 committed:
fix(android): compatible HashMap and ArrayList for ArgumentUtils

wwwcg committed:
fix(ios): fix timing issue of backgroundImage prop

wwwcg committed:
test(ios): add unit test infrastructure for ios project

wwwcg committed:
fix(ios): solve some problems of podspec (#3789)

* fix(ios): solve some problems of podspec

such as normal pod integration is not supported, search path errors, etc

* chore(devtools): remove the base64 dependency

* chore(ios): update json subspec config in podfile

gguoyu committed:
feat(vue-next): compatible vue2 component tag name format

zealotchen0 committed:
feat(vue-next): fix hippytext setText (#3790)

Co-authored-by: OpenHippy <124017524+open-hippy@users.noreply.github.com>

zealotchen0 committed:
feat(vue-next): fix image url style parse (#3791)

Co-authored-by: OpenHippy <124017524+open-hippy@users.noreply.github.com>

wwwcg committed:
build(devtools): fix build error of conversion warnings in target deps

siguangli committed:
feat(android): support fcp monitor (#3796)

* feat(android): support fcp monitor

* fix(android): rename to PAINT_TYPE_KEY

---------

Co-authored-by: maxli <maxli@tencent.com>

siguangli2018 committed:
feat(android): add BaseEngineContext interface

wwwcg committed:
fix(ios): fix conversion crash from oc object to CtxValue

wwwcg committed:
fix(ios): add ResponseSenderBlock type support for component parameter

wwwcg committed:
chore(ios): eliminate warnings and optimize code (#3798)

* chore(ios): eliminate warnings and optimize code

* chore(ios): clean up obsolete functions

wwwcg committed:
fix(ios): thread race of componentDataByName in UIManager (#3801)

* fix(ios): thread race of componentDataByName in UIManager

* fix(ios): thread race of componentDataByName in UIManager2

wwwcg committed:
docs(ios): add iOS 3.0 upgrade guide (#3757)

* docs(ios): add iOS 3.0 upgrade guide

* docs(ios): add iOS 3.0 upgrade guide

siguangli committed:
Feat/3.0 upgrade guidelines (#3805)

* refactor(docs): update 3.0 upgrade guidelines

* refactor(docs): update iOS 3.0 upgrade guidelines

---------

Co-authored-by: maxli <maxli@tencent.com>

siguangli2018 committed:
refactor(docs): update 3.0 upgrade guidelines and privacy

siguangli2018 committed:
refactor(docs): fix markdownlint error

wwwcg committed:
feat(ios): add fcp monitor support (#3803)

* feat(ios): add fcp monitor support

* chore(ios): performance related code optimize

* test(ios): add some test code

zealotchen0 committed:
feat(doc): 3.0 performance api doc update

wwwcg committed:
docs(ios): update upgrade guidelines of iOS

zealotchen0 committed:
feat(doc): fix docs markdown lint

siguangli committed:
feat(android): refactoring WaterfallView (#3809)

* feat(android): refactoring WaterfallView

* fix(vue3): fix vue3 refreshing && vue2 style

* fix(android): fix PullRefreshContainer size for recycler view

* fix(android): add copyright header for ItemLayoutParams

---------

Co-authored-by: maxli <maxli@tencent.com>
Co-authored-by: zealotchen <zealotchen@tencent.com>

zealotchen0 committed:
feat(js): remove demo clipboard

zealotchen0 committed:
chore(release): modify npm publish config (#3815)

* chore(release): modify npm publish config

* chore(doc): update 3.0 vue doc

siguangli committed:
chore(action): add project_artifact_snapshot.yml (#3817)

Co-authored-by: maxli <maxli@tencent.com>

siguangli committed:
fix(action): change project_artifact_snapshot name (#3818)

Co-authored-by: maxli <maxli@tencent.com>

siguangli2018 committed:
docs(android): update android snapshot 3.2.0 version

wwwcg committed:
chore(ios): fix home page layout and update default demo bundle (#3814)

wwwcg committed:
refactor(ios): fix dependency issues between modules and podspec error (#3819)

* chore(ios): fine-tune the dependencies between modules

* refactor(ios): fix dependency issues between modules

* fix(ios): fix podspec lint issues

wwwcg committed:
fix(ios): tiny issue of  TransactionListener in UIManager (#3821)

siguangli2018 committed:
refactor(docs): add six key point

siguangli committed:
fix(android): update react demo bundle (#3825)

Co-authored-by: maxli <maxli@tencent.com>

zealotchen0 committed:
feat(dom): fix vue3 list item position  (#3826)

* feat(dom): fix vue3 list sort

* feat(dom): fix vue3 list item position

* feat(dom): perf code && adjust needSortByIndex

* feat(dom): fix dom manager unit test

* feat(dom): avoid get self depth when skip sortByIndex

zealotchen0 committed:
feat(vue): fix vue replace &nbsp

wwwcg committed:
chore(ios): improve parameter mismatch tips

wwwcg committed:
refactor(ios): Improve compatibility of imageProvider

wwwcg committed:
refactor(ios): increase robustness of HippyUrl process

gguoyu committed:
fix(vue-next): fix vue-next package dependencies issue

* fix(vue-next): fix vue package dependencies issue

* fix(vue-next): fix vue-next package dependencies issue

zealotchen0 committed:
feat(doc): update hippy3.0 vue-next version

wwwcg committed:
feat(ios): unify HippyBridge's notifications and improve compatibility

wwwcg committed:
feat(ios): add custom image loader support (#3830)

* feat(ios): add custom image loader support

add HippyImageCustomLoaderProtocol

* refactor(ios): refactor frame structure

* chore(ios): remove obsolete comment

wwwcg committed:
chore(ios): add SecondaryBundleDidLoadNotification for compatibility

wwwcg committed:
chore(ios): optimize parameter check tips

wwwcg committed:
fix(ios): move nodes op non same parent support

zealotchen0 committed:
feat(vue-next): add vue3 css cache

wwwcg committed:
fix(ios): fix move node index bug

wwwcg committed:
fix(ios): debug reload module not working

wwwcg committed:
chore(ios): eliminate some warnings

wwwcg committed:
fix(ios): eliminate Devtools connection warning

wwwcg committed:
fix(ios): image capInsets and getSize's bug

Stay compatible with hippy2

wwwcg committed:
fix(ios): lineHeight attribute was incorrectly inherited

wwwcg committed:
fix(dom): missing render_info update before move operation

wwwcg committed:
fix(ios): add null protection before invalid move operation

wwwcg committed:
fix(dom): optimized results not used in move operation

siguangli2018 committed:
fix(android): support use custom DisplayMetrics

zealotchen0 committed:
feat(vue-next): update comments

siguangli committed:
fix(android): debugServerHost invalid setting (#3841)

Co-authored-by: siguangli <maxli@tencent.com>

zealotchen0 committed:
feat(vue-next): add update event operation (#3837)

* feat(vue-next): add update event operation

* feat(vue-next): update event comments

wwwcg committed:
fix(ios): add protection against null value exceptions

siguangli committed:
fix(android): getScreenShot bitmap width or height < 0 (#3842)

Co-authored-by: siguangli <maxli@tencent.com>

wwwcg committed:
refactor(ios): non-standard url support of bundle loading

wwwcg committed:
fix(ios): multi-threading issue in createView

and resolve minor issue of duplicate assignment

zealotchen0 committed:
feat(action): update v8 compiler action

zealotchen0 committed:
feat(action): fix v8 build error

siguangli2018 committed:
fix(android): correct move node impl with different pid

siguangli2018 committed:
fix(android): image span draw transparent color when no src set

siguangli2018 committed:
refactor(docs): add android snapshot version description

wwwcg committed:
chore(build): update husky commit hook command

wwwcg committed:
fix(ios): null exception when converting JSI object to dictionary

wwwcg committed:
test(ios): add xctestplan file

wwwcg committed:
build(ios): remove xcodegen usage due to test support issues

wwwcg committed:
fix(ios): error in move node operation of ViewPager

wwwcg committed:
feat(ios): update ViewPager's pageIndex automatically after data changes (#3857)

* feat(ios): pageIndex of ViewPager auto update after data changes

Also fixed the issue of missing onPageSelected callbacks upon first entry;

The logic for automatic updates keep same with Android, as follows
1. If the previous item only changes its location,
update the current location and keep the current item displayed.
2. If the previous item does not exist, do not adjust the position,
but keep the current position in the valid range (that is, 0 ~ count-1).

* fix(ios): auto-update of page index may lag by one frame in view pager

this is a minor fix for feat 'pageIndex of ViewPager auto update after data changes'

siguangli committed:
feat(android): add orientationChange event for Modal (#3861)

Co-authored-by: siguangli <maxli@tencent.com>

zealotchen0 committed:
fix(devtool): fix muli devtool instance crash on android (#3862)

* fix(devtool): fix muli devtool instance crash on android

* fix(devtool): modify some comment

zealotchen0 committed:
feat(action): install ninja-build before v8 build (#3864)

* feat(action): fix v8 build error

* feat(action): fix v8 compiler action

* feat(action): fix v8 compiler action

* feat(action): fix v8 compiler action

wwwcg committed:
build(ios): fix workflow build script

zealotchen0 committed:
feat(vue): fix attribute inheritance style not empty (#3868)

zealotchen0 committed:
feat(vue-next): modify comment

zealotchen0 committed:
feat(v8): add version log for v8

wwwcg committed:
chore(ios): add addFrameCallback method for devtools to avoid warning

Note that the implementation of this method is empty

wwwcg committed:
docs(privacy): update privacy guide

wwwcg committed:
fix(ios): add protection for targetContentOffset outliers in ViewPager

Commit:
fix(ios): animation nullptr crash

wwwcg committed:
docs(privacy): add privacy guide for developer

wwwcg committed:
refactor(ios): revert hippyShadowView to shadowView for compatibility

Important changes:
Since Hippy3 changed the name of the method used to get shadowView early on,
To maintain compatibility with hippy2, modify it here back to shadowView

siguangli committed:
fix(android): should updateExtra after create host view (#3879)

Co-authored-by: siguangli <maxli@tencent.com>

wwwcg committed:
refactor(ios):  add method call parameter matching protection

To prevent wild pointer problem caused by the number of parameters
passed by the front end being smaller than the actual parameters

wwwcg committed:
chore(ios): eliminate unnecessary null warning

wwwcg committed:
refactor(ios):  add method call parameter matching protection 2

wwwcg committed:
fix(ios): dangling pointer in HippyBridge when unload after invalidate

siguangli committed:
refactor(docs): update screenshot and 3.0 upgrade guidelines md (#3883)

Co-authored-by: maxli <maxli@tencent.com>

wwwcg committed:
fix(ios): resolve null pointer crash when loadBundleURL

wwwcg committed:
fix(ios): resolve non-standard object type when creating js object

etkmao committed:
fix(perf): fix OnResourceLoadEnd crash (#3888)

siguangli committed:
Feat/screen snapshot (#3884)

* fix(android): use screen density for dp to px

* feat(android): support get screenshot bitmap from hippy engine

---------

Co-authored-by: siguangli <maxli@tencent.com>

etkmao committed:
Bugfix/anim state crash (#3891)

* fix(perf): animation state not crash on release version

* fix(core): crash protect when onInstanceDestroy is invoked 2 timers

* fix(core): animation state not crash on release version

wwwcg committed:
fix(ios): loss of touch end or cancel event in multi-finger scenarios (#3892)

* fix(ios): loss of touch end or cancel event in multi-finger scenarios

In a multi-finger scenario,
the _touchBeganView local variable only records the last touch view
since touch began is entered multiple times,
causing the began and cancel/end events to be unmatched.

To fix this, we use a simple approach -
that record all touch began Views and send events to all recorded views at the end.

Please note that we have not fully adapted the multi-fingered scenario.

* fix(ios): resolve exception where touchView may be nil

siguangli committed:
refactor(docs): update 3.0 upgrade guidelines md (#3894)

Co-authored-by: siguangli <maxli@tencent.com>

wwwcg committed:
refactor(ios): clean and refactor the HippyBridge's code

etkmao committed:
fix(perf): fix OnResourceLoadEnd crash (#3895)

wwwcg committed:
fix(ios): add CSS name matching semantics to fontFamily of TextInput

1780243524 committed:
fix(android): resolve local ref overflow

wwwcg committed:
refactor(ios): add gesture event delegate for hippyBridge

useful for external data reporting of hippy gesture events

etkmao committed:
fix(core): fix android scope crash (#3906)

xiezheng-XD committed:
docs(vue): typo in vue's README.md (#3444)

siguangli committed:
refactor(docs): update 3.0 upgrade guidelines md (#3908)

Co-authored-by: siguangli <maxli@tencent.com>

zealotchen0 committed:
feat(doc): update js doc

wwwcg committed:
feat(ios): add paragraph styles support for TextInput

TextInput Current support:
1.lineHeight
2.lineSpacing
3.lineHeightMultiple

siguangli committed:
fix(android): remove onSystemUiVisibilityChange listener (#3913)

Co-authored-by: siguangli <maxli@tencent.com>

wwwcg committed:
fix(ios): potential thread race in HippyJSExecutor

wwwcg committed:
fix(ios): performance api of FP not working

zealotchen0 committed:
feat(doc): update js doc for 3.3

wwwcg committed:
docs(ios): update integration and upgrade guildlines

wwwcg committed:
chore(ios): resolve podspec issues

remove use_frameworks env variable

wwwcg committed:
chore(ios): resolve podspec lint issue

zealotchen0 committed:
feat(react): add font style inherit

wwwcg committed:
chore(ios): update integration guide

zealotchen0 committed:
feat(vue): fix default native style is empty

siguangli committed:
fix(android): different sticky item position crash (#3930)

Co-authored-by: maxli <maxli@tencent.com>
Co-authored-by: OpenHippy <124017524+open-hippy@users.noreply.github.com>

siguangli committed:
feat(doc): update 3.3.0 integration guidelines (#3931)

Co-authored-by: maxli <maxli@tencent.com>

wwwcg committed:
fix(ios): responsable region issue of Textinput (#3932)

* fix(ios): responsable region issue of Textinput

* chore(ios): remove unused code in TextView

siguangli committed:
feat(android): text input support lineHeight (#3936)

Co-authored-by: maxli <maxli@tencent.com>

zealotchen0 committed:
chore(doc): update hippy 3.3.0 doc

wwwcg committed:
fix(ios): header and footer refresh bugs (#3940)

* fix(ios): header and footer refresh bugs

* chore(ios): refactor some code in HippyRefresh

wwwcg committed:
fix(ios): eliminate all warnings and open treat_warning_as_error

zealotchen0 committed:
fix(doc): fix markdown lint

wwwcg committed:
feat(ios): add app bundle and container path support for hpfile url

siguangli committed:
feat(android): vfs support custom image loader (#3947)

Co-authored-by: maxli <maxli@tencent.com>
Co-authored-by: OpenHippy <124017524+open-hippy@users.noreply.github.com>

siguangli committed:
fix(android): font weight compatible with version 2.x (#3948)

Co-authored-by: maxli <maxli@tencent.com>

wwwcg committed:
feat(ios,android): add request duration in fetch response header

zealotchen0 committed:
feat(js): fix waterfall for iOS and ohos

siguangli2018 committed:
fix(android): erase NativeRenderManager when destroy

siguangli2018 committed:
fix(android): to use WeakReference as much as possible

siguangli2018 committed:
fix(android): change to using global image cache

siguangli2018 committed:
fix(android): removeRootView params may be null

siguangli2018 committed:
fix(android): sticky item type calculate using node id

wwwcg committed:
fix(ios, android): performance records inaccurate issues

siguangli2018 committed:
fix(android): lru cache is thread safe

zealotchen0 committed:
fix(devtools): fix vue3 devtools element

zealotchen0 committed:
feat(demo): update demo dependency

siguangli committed:
fix(android): add custom display metrics (#3965)

Co-authored-by: maxli <maxli@tencent.com>
Co-authored-by: OpenHippy <124017524+open-hippy@users.noreply.github.com>

lvfen committed:
fix(voltron): fix voltron http request dio null error (#3962)

* fix(voltron): resolve remoteServerUrl parse version id failed

* fix(voltron): add license for flutter_render_test.dart

* fix(voltron): remoteServerUrl connect without path

* fix(voltron): remoteServerUrl connect without path

* fix(voltron): remoteServerUrl connect without path

* fix(voltron): fix voltron http request dio null error

* fix(voltron): fix voltron http request dio null error

---------

Co-authored-by: OpenHippy <124017524+open-hippy@users.noreply.github.com>

etkmao committed:
fix(core): check crash on config SetScaleFactor (#3967)

wwwcg committed:
fix(ios): iOS18 adaptation of TextView component

siguangli committed:
fix(android): use weak reference replace strong (#3975)

Co-authored-by: maxli <maxli@tencent.com>

wwwcg committed:
fix(ios): add exception handle process for jsc (#3976)

* fix(ios): add exception handle process for jsc

* chore(ios): remove unnecessary exception check log

wwwcg committed:
chore(ios): eliminate code warnings

siguangli committed:
fix(android): onGlobalLayout use root view size (#3982)

Co-authored-by: maxli <maxli@tencent.com>

wwwcg committed:
refactor(ios): remove unused data structure in waterfall item

wwwcg committed:
refactor(ios): subcomponents to hippySubviews prop for compatibility

Earlier Hippy3 versions used subcomponents to represent children,
For better compatibility with Hippy2, change back to hippySubviews

wwwcg committed:
fix(ios): remove invalid if statement at cell reuse

siguangli2018 committed:
fix(android): add buildTruncateAtEndStaticLayout impl

siguangli2018 committed:
fix(android): delete virtual child node when parent deleted

siguangli2018 committed:
fix(android): add ControllerRegistry destroy

wwwcg committed:
fix(ios): improved accuracy of ViewPager's ScrollStateChanged API

wwwcg committed:
fix(ios): resolve floating point size comparison issues (#3989)

wwwcg committed:
refactor(ios): remove all context wrapper classes (#3994)

zealotchen0 committed:
feat(vue-next): fix onInterceptTouchEvent not work (#3995)

* feat(vue-next): fix onInterceptTouchEvent not work

* fix(vue-next): fix remove attr not update

* fix(vue-next): revert vue-next version

siguangli2018 committed:
fix(android): append init props for image load

siguangli2018 committed:
fix(android): use Spanned to getSpans

zealotchen0 committed:
fix(vue-next): fix beforeLoadStyle not work

wwwcg committed:
fix(ios): resolve reload not working (#4006)

refactor hippy demo code

wwwcg committed:
chore(ios): update demo and eliminate some warnings

wwwcg committed:
fix(ios): resolve conversion error between NSString and CtxValue (#4007)

add unit test for data conversion

wwwcg committed:
fix(ios): onShow event of modal component not working (#4012)

remove obsolete onLayout property of shadowView

wwwcg committed:
refactor(ios): remove unused code in scope and update demo (#4013)

siguangli2018 committed:
fix(android): add try catch for vfs processor chain get

wwwcg committed:
fix(ios): null pointer exception of bridge setup in rare scenarios

wwwcg committed:
build(voltron): resolve voltron build error

etkmao committed:
fix(ios): fix object_data_map crash (#4017)

when jsc_vm released before jsc_ctx

wwwcg committed:
fix(ios): resolve engine reuse issue (#4019)

refactor UIManager and HippyBridge,
remove some redundant UIView's category

etkmao committed:
refactor(ios): avoid mutex crash in special app environment (#4020)

* fix(ios): check static invalid mutex crash

* fix(ios): move static maps to heap for checking static overflow crash

wwwcg committed:
fix(ios): resolve jsc_try_catch invalid issue (#4021)

wwwcg committed:
refactor(ios): add a uiManager getter for UIView (#4022)

For compatibility,
some third-party components want to get UIManager
or HippyBridge directly through UIView,
although there is a slight performance penalty doing this.

gguoyu committed:
fix(vue-next): fix vuejs ugrade issue (#4004)

* fix(vue-next): fix vuejs ugrade issue

1. fix vuejs 3.4.32 patchProps API change issue
2. compatible vuejs compiler-core API change issue
3. upgrade vue-next demo package version

* fix(vue-next): fix vue-next-ssr-demo issue

1. compatible vue upgrade issue
2. fix vue-next-ssr demo issue

wwwcg committed:
fix(ios): resolve hot reload issue (#4024)

Use the default network loading timeout
Delete some obsolete demo classes

wwwcg committed:
fix(ios): resolve sandbox dir issue in debug mode (#4025)

wwwcg committed:
fix(ios): keyboard related events not working for TextView (#4026)

add missing onKeyboardHeightChanged property

wwwcg committed:
fix(ios): onAttachedToWindow event not triggered (#4027)

wwwcg committed:
refactor(ios): rewrite bundle loading process (#4028)

wwwcg committed:
fix(ios): accuracy of ViewPager's onPageScroll event parameters
fix(ios): improve accuracy of ViewPager's onPageScroll parameters

wwwcg committed:
fix(ios): avoid startup crash when script load error

wwwcg committed:
feat(ios): modular header import support

wwwcg committed:
fix(ios): potential thread race while accessing bridge

Refactoring: remove some debug code

zealotchen0 committed:
feat(vue): support ul refresh scroll (#4032)

* feat(vue-next): fix beforeLoadStyle not work

* feat(vue): support_ul_refresh_scroll

zealotchen0 committed:
feat(web): fix dymamic load for web render
